disnake/components.py

```diff
@@ -376,6 +382,86 @@ def to_dict(self) -> SelectOptionPayload:
         return payload
 
 
+class TextInput(Component):
+    """Represents an input text from the Discord Bot UI Kit.
+
+    This can only be used in a :class:`~.ui.Modal`.
+
+    .. versionadded:: 2.4
+
+    .. note::
+
+        The user constructible and usable type to create an input text is
+        :class:`disnake.ui.TextInput`, not this one.
+
+    Attributes
+    -----------
+    style: :class:`TextInputStyle`
+        The style of the input text.
+    label: Optional[:class:`str`]
+        The label of the input text.
+    custom_id: :class:`str`
+        The ID of the input text that gets received during an interaction.
+    placeholder: Optional[:class:`str`]
+        The placeholder text that is shown if nothing is entered.
+    value: Optional[:class:`str`]
+        The pre-filled text of the input text.
+    required: :class:`bool`
+        Whether the input text is required. Defaults to ``True``.
+    min_length: :class:`int`
+        The minimum length of the input text. Defaults to ``0``.
+    max_length: Optional[:class:`int`]
+        The maximum length of the input text.
+    """
+
+    __slots__: Tuple[str, ...] = (
+        "style",
+        "custom_id",
+        "label",
+        "placeholder",
+        "value",
+        "required",
+        "max_length",
+        "min_length",
+    )
+
+    __repr_info__: ClassVar[Tuple[str, ...]] = __slots__
+
+    def __init__(self, data: TextInputPayload) -> None:
+        style = data.get("style", TextInputStyle.short.value)
+
+        self.type: ComponentType = try_enum(ComponentType, data["type"])
+        self.custom_id: str = data["custom_id"]
+        self.style: TextInputStyle = try_enum(TextInputStyle, style)
+        self.label: Optional[str] = data.get("label")
+        self.placeholder: Optional[str] = data.get("placeholder")
+        self.value: Optional[str] = data.get("value")
+        self.required: bool = data.get("required", True)
+        self.min_length: int = data.get("min_length", 0)
+        self.max_length: Optional[int] = data.get("max_length")
+
+    def to_dict(self) -> TextInputPayload:
+        payload: TextInputPayload = {
+            "type": self.type.value,
+            "style": self.style.value,  # type: ignore
+            "label": self.label,  # type: ignore
+            "custom_id": self.custom_id,
+            "required": self.required,
+            "min_length": self.min_length,
```

Comment:
this should be optional IMO

Reply:
Several of these are optional, according to the API docs

Reply:
Fixed a portion of this.

Reply:
> this should be optional IMO

I only used the default values the API uses. `required` defaults to True and `min_length` defaults to 0